tests/kubernetes/conftest.py

```diff
@@ -0,0 +1,110 @@
+from tests.kubernetes.minikube import *
+from tests.kubernetes.utils import *
+import docker
+import pytest
+
+
+@pytest.fixture(scope="module")
+def local_registry(request):
+    client = docker.from_env(version='auto')
+    final_agent_image_name = request.config.getoption("--k8s-agent-name")
+    final_agent_image_tag = request.config.getoption("--k8s-agent-tag")
+    try:
+        final_image = client.images.get(final_agent_image_name + ":" + final_agent_image_tag)
+    except:
+        try:
+            print("\nAgent image '%s:%s' not found in local registry." % (final_agent_image_name, final_agent_image_tag))
+            print("\nAttempting to pull from remote registry ...")
+            final_image = client.images.pull(final_agent_image_name, tag=final_agent_image_tag)
+        except:
+            final_image = None
+    assert final_image, "agent image '%s:%s' not found!" % (final_agent_image_name, final_agent_image_tag)
+    try:
+        client.containers.get("registry")
+        print("\nRegistry container localhost:5000 already running")
+    except:
```

Comment:
Avoid pokemon exception handlers, it makes it frustrating when running tests locally by preventing `Ctrl-C` from exiting the process if pressed when this code is executing.  It also masks other problems besides the container just not existing.